In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm

#### 1. Complete the ANOVA table. Write all numbers with NO decimal place except p-value. p-value should be written in 5 decimal places.

In [2]:
n = 10
r = -0.9212
s = 124.3
s2 = 124.3**2
SSE = s2*(n-2)
SST = -SSE/((r)**2-1)
SSR = SST - SSE

k = 1
df = n - k - 1

MSR = SSR/k
MSE = SSE/df
F0 = MSR/MSE

pvalF = 1-stats.f.cdf(F0,k,df)

pd.options.display.float_format = '{:,.5f}'.format
ANOVA = pd.DataFrame({
    'Source': ['SSR', 'SSE', 'SST'],
    'SS': [SSR, SSE, SST],
    'deg of freedom': [k, df, ''],
    'MS': [MSR, MSE, ''],
    'F0': [F0, '', ''],
    'pvalue': [pvalF, '', '']
})
ANOVA

Source            SS deg of freedom            MS       F0  pvalue
0    SSR 692,853.32806              1 692,853.32806 44.84345 0.00015
1    SSE 123,603.92000              8  15,450.49000                 
2    SST 816,457.24806

#### 2. Complete the following chart. Write all numbers with ONE decimal place except p-value. p-value should be written in 5 decimal places.

In [3]:
sebeta1hat = 5.4
beta0hat = 2594.7
Sxx = (s/sebeta1hat)**2
beta1hat = -np.sqrt(SSR/Sxx)
Sxy = beta1hat * Sxx
Syy = SST

ybar = 21121/n
xbar = (beta0hat-ybar)/(-beta1hat)

beta1hat = Sxy/Sxx
beta0hat = ybar-beta1hat*xbar

sebeta1hat = s/np.sqrt(Sxx)
sebeta0hat = s*np.sqrt(1/n + xbar**2/Sxx)
tbeta1 = beta1hat/sebeta1hat
tbeta0 = beta0hat/sebeta0hat

pvalue1 = (1-stats.t.cdf(abs(tbeta1), df))*2
pvalue0 = (1-stats.t.cdf(abs(tbeta0), df))*2

alpha = 0.05
UBbeta1 = beta1hat + stats.t(df=df).ppf(1-alpha/2)*sebeta1hat
LBbeta1 = beta1hat - stats.t(df=df,).ppf(1-alpha/2)*sebeta1hat
UBbeta0 = beta0hat + stats.t(df=df).ppf(1-alpha/2)*sebeta0hat
LBbeta0 = beta0hat - stats.t(df=df,).ppf(1-alpha/2)*sebeta0hat

paraEst = pd.DataFrame({
    'Parameter': ['intercept', 'slope'],
    'Estimates': [beta0hat, beta1hat],
    's.e.': [sebeta0hat, sebeta1hat],
    't-value': [tbeta0, tbeta1],
    'p-value': [pvalue0, pvalue1],
    'LB95': [LBbeta0, LBbeta1],
    'UB95': [UBbeta0, UBbeta1]
    
})
paraEst

Parameter   Estimates     s.e.  t-value  p-value        LB95        UB95
0  intercept 2,594.70000 82.08979 31.60807  0.00000 2,405.40060 2,783.99940
1      slope   -36.16124  5.40000 -6.69653  0.00015   -48.61366   -23.70882

#### 3. Write the equation for the regression line of above data (one decimal place).

$y_i$= 2594.7 - 36.2 $x_i$

#### 4. Find $R^2$ (three decimal place). What is the meaning of the $R^2$ value in the context of the problem?

In [4]:
R2 = r**2
R2

0.84860944

84.9% of the variation of y′s is explained by the regression model.

#### 5. Find the residual of the first observation $x_1=15.5,\,\,\,$ $y_1=2158$.

In [5]:
yihat = beta0hat + beta1hat*(15.5)
ei = 2158-yihat
ei

123.79918191856905

#### 6. Find the 95% confidence interval for $\hat\beta_1$ (one decial place).

(-48.6, -23.7)